### PR TITLE
Simplifying check for connection validator

### PR DESCRIPTION
**Pull request by saratvemulapalli**
Signed-off-by: Sarat Vemulapalli <vemulapallisarat@gmail.com>

### Description
While reading up code for connection validator to fix a bug in [Feature/extensions]: https://github.com/opensearch-project/OpenSearch/issues/2772. Felt like the code could be made little simpler and readable.
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: server/src/main/java/org/opensearch/transport/TransportService.java [hunk 1]
```diff
@@ -431,10 +431,14 @@ public void connectToNode(final DiscoveryNode node, ConnectionProfile connection
...
-                final DiscoveryNode remote = resp.discoveryNode;
-                if (node.equals(remote) == false) {
-                    throw new ConnectTransportException(node, "handshake failed. unexpected remote node " + remote);
+            handshake(newConnection, actualProfile.getHandshakeTimeout().millis(), cn -> true, ActionListener.map(listener, response -> {
+                final DiscoveryNode responseNode = response.discoveryNode;
+                /**
+                 * Validate if the node we are trying to connect to and the response recieved remote node is the same.
+                 * If not, error out.
+                 */
+                if (!node.equals(responseNode)) {
```
Comment (reta) [inline server/src/main/java/org/opensearch/transport/TransportService.java:440]: Sorry, I didn't get the change - isn't it the same as `node.equals(remote) == false` ? (I think @nknize prefers explicit `false` checks)

Comment (saratvemulapalli) [reply server/src/main/java/org/opensearch/transport/TransportService.java:440]: It felt more readable. 
But if we prefer the earlier, I am ok with it :) 

Thanks for the review @reta!

Comment (saratvemulapalli) [reply server/src/main/java/org/opensearch/transport/TransportService.java:440]: Do you still feel the value for the comments ? 
If not I can close out the PR.

Comment (reta) [reply server/src/main/java/org/opensearch/transport/TransportService.java:440]: To be fair, the previous version with `remote` was kind of explicit - the `node` should not be the same as `remote` one ...

**Pull request closed without merging**